hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupInfoManagerImpl.java

```diff
@@ -1227,9 +1227,13 @@ public synchronized void renameRSGroup(String oldName, String newName) throws IO
       throw new ConstraintException(RSGroupInfo.DEFAULT_GROUP + " can't be rename");
     }
     checkGroupName(newName);
-
+    //getRSGroupInfo validates old RSGroup existence.
     RSGroupInfo oldRSG = getRSGroupInfo(oldName);
     Map<String, RSGroupInfo> rsGroupMap = holder.groupName2Group;
+    if (rsGroupMap.get(newName) != null) {
```

Comment:
`rsGroupMap.containsKey()` is a better check.

Reply:
handled, used rsGroupMap.containsKey()